map_gui/src/lib.rs

```diff
@@ -0,0 +1,342 @@
+//! Several distinct tools/applications all share the same general structure for their shared GUI
+//! state, based around drawing and interacting with a Map.
+
+use abstutil::{CmdArgs, Timer};
+use geom::{Circle, Distance, Duration, Pt2D, Time};
+use map_model::{IntersectionID, Map};
+use sim::Sim;
+use widgetry::{EventCtx, GfxCtx, SharedAppState, State, Transition, Warper};
+
+use crate::helpers::ID;
+use crate::render::{DrawOptions, Renderable};
+use colors::{ColorScheme, ColorSchemeChoice};
+use options::Options;
+use render::DrawMap;
+
+pub mod colors;
+pub mod common;
+pub mod game;
+pub mod helpers;
+pub mod load;
+pub mod misc_tools;
+pub mod options;
+pub mod render;
+
+/// Why not use composition and put the Map, DrawMap, etc in a struct? I think it wouldn't let us
+/// have any common widgetry States... although maybe we can instead organize the common state into
```

Comment:
> wouldn't let us have any common widgetry States

What do you mean by "common widgetry states"?

Reply:
Things like `CityPicker`, which is a `widgetry::State`. This comment was a rambling alternative considered to the current design; I'll clean the doc comments for this.